### ISSUE TITLE
Set up relative pathing for images and various resources

### ISSUE THREAD

**Issue by kmfarley11** [Triangle345/GT#13]
As a game engine, users will be adding many files: sprites, tile-sets, models, audio, scripts, etc....

Go uses pathing relative to where the project is run from. So running TestGame.go from within C:\AnyIrrelevantFolder will not work. 

We should define our resource locations and enact a relative (project) pathing module from within the engine. Leaving the code as-is opens up potential hazards when users run from command line or move files around. It would also be cleaner and easier to develop with defined, relative pathing.

**Comment by Triangle345** [comment]
I agree. We should have a designated "project" folder.  I propose writing a tool that will create a "new project" for a user; thus setting everything up including all paths.  This tool could set up the directory structure and a necessary "main" file that is used to run the game.  This directory structure would include images folder, fonts folder, models, etc.
This tool could also setup gopath for the project, this way the user can create it anywhere really.  This would work really well if we ever want to create a GUI editor.

**Comment by kmfarley11** [comment]
I think that the project folder is a good idea, but that should be tagged into a deployment issue. For the  immediate fix of this bug, in order to clean up our engine resources, we should probably just make an EngineImages or ExampleImages folder in our Assets directory. So that we do not have to have to make test.png appear in every directory that we create an example in

**Comment by kmfarley11** [comment]
I redact part of my last comment. **I do think that this issue should reference the project folder implementation** (as you mentioned) according to the original description and comment. However, I originally created the issue because of a bug I encountered. The bug involves adding images to the aggregate and using them. After digging I have found the problem and will create a new Issue and tag it as a bug. Then I will remove the bug tag from this one and add the Deployment tag.

In other words, this issue should be referenced to set up project creation and relative pathing according to that. The bug I found will be addressed in a separate issue.

**Comment by kmfarley11** [comment]
Yes, we should have some sort of project creation and organization. A simple tool / utility which keeps track of all that would be quite useful. Once implemented we need to ensure that the AggregateImage loads in images according to the project we generate. This is referenced in #24.

**Comment by Triangle345** [comment]
A new issue will need to be created later for a new project tool, will probably be part of the editor gui.

**Comment by Triangle345** [comment]
i want to modify this a bit and maybe in GT package add a function like GetImage(path) which would automatically return the absolute path, err combo based on AssetsImages so only thing the user has to type GetImage("test.png") etc. Same for models and fonts.
